File: benchmark/output/python-pygame/gemini/countdown_timer.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Countdown Timer")
clock = pygame.time.Clock()
font = pygame.font.Font(None, 72)

# Game state variables
timer = 30.0

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    # Update logic
    if timer > 0.0:
        timer -= dt
        if timer < 0.0:
            timer = 0.0

    # Draw logic
    screen.fill((0, 0, 0))
    
    if timer > 0.0:
        text_surface = font.render(str(int(timer)), True, (255, 255, 255))
        screen.blit(text_surface, (350, 280))
    else:
        text_surface = font.render("Time Up!", True, (255, 255, 255))
        screen.blit(text_surface, (330, 280))

    pygame.display.flip()

pygame.quit()
sys.exit()

File: benchmark/output/lua-love/gemini/countdown_timer.lua
local timer = 30.0

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Countdown Timer")
end

function love.update(dt)
    timer = timer - dt
    if timer < 0.0 then
        timer = 0.0
    end
end

function love.draw()
    if timer > 0.0 then
        love.graphics.print(tostring(math.ceil(timer)), 350, 280)
    else
        love.graphics.print("Time Up!", 330, 280)
    end
end